File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       map<int,int> ans;
        for(int i=0;i<nums.size();i++){
            if(ans.count(target-nums[i]))
                return {ans[target-nums[i]],i};
            ans[nums[i]]=i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
       Dictionary<int,int> ans=new Dictionary<int,int>();
       for(int i=0;i<nums.Length;i++){
           if(ans.TryGetValue(target-nums[i],out int index)){
               return new int[]{i,index};
           }else ans[nums[i]]=i;
       }
       return new int[]{};
    }
}